kythe/java/com/google/devtools/kythe/analyzers/java/KytheTreeScanner.java

```diff
@@ -961,7 +963,11 @@ public JavaNode visitTypeArray(JCArrayTypeTree arrayType, TreeContext owner) {
   public JavaNode visitAnnotation(JCAnnotation annotation, TreeContext owner) {
     TreeContext ctx = owner.down(annotation);
     scanList(annotation.getArguments(), ctx);
-    return scan(annotation.getAnnotationType(), ctx);
+    boolean savedInRefIdContext = inRefIdContext;
+    inRefIdContext = true;
+    JavaNode node = scan(annotation.getAnnotationType(), ctx);
```

Comment:
I don't think it's possible to have an annotation scoped "inside" an annotation? I guess the saving/restoring is the canonical pattern to use, though.